### ISSUE TITLE
Worksheet drilldown improvements

### ISSUE THREAD

**Issue by pmarko1711** [data-dot-all/dataall#984]
**Is your idea related to a problem? Please describe.**
A worksheet belongs to a team (at least it gets created so in `console/worksheets/new`). Selecting a particular existing worksheet I get to a page `console/worksheets/[id]/`. Once I'm there I need to select Environment, Team, Database. This gets a bit confusing when my user is a member of multiple teams and with multiple shares existing. 

**Describe the solution you'd like**
To make it less so (and only if it make sense):

1.  Perhaps here the most relevant environment (if that can be guessed) could be preselected, same for the team. In fact, it is a bit confusing to me why I can even select an environment & different to the one the worksheet was created by

2. Only the combinations that would work for my user should be shown in the dropdowns. It appears to me that when selecting a team, I see sometimes more shares than should be the case. Since I dropdown for database gets enabled only when I select a team, I should only be shown databases relevant for the selected team, or (I sometimes seem to see shares belonging to different teams)?

**Comment by noah-paige** [comment]
Hi @pmarko1711 - thanks for opening this issue. I definitely agree that selecting a team separate for the one you created the Worksheet with is confusing and would be inclined to remove it. In that case:

- Data.all User Creates a Worksheet Specifying a Team to Create Worksheet With
- On Opening the Worksheet User Can Select Environment, Database, and Table (Team is fixed as the one that created the worksheet)
  - With a Default Environment Selected

For the issue with showing shares from other groups - I think it relates to this [outstanding issue](https://github.com/data-dot-all/dataall/issues/805) as well. Where a user belonging to more than TeamA and TeamB can see shares for both teams rather than just shares for the Team that created the Worksheet

Do let me know if the above aligns with your thinking as well - from there we prioritize a time for the fix for the above

**Comment by zsaltys** [comment]
@noah-paige I think this suggestion makes a lot of sense. I agree that worksheets should be tied to a team. I would go further and lock environment as well as part of sheet creation.

Let's think about it. Why do we have worksheets as savable entities? It is so that teams could share them and rerun the same saved query. So it doesn't make sense that once a team who owns a sheet needs to figure out for what db or environment this saved query was configured. So when I SAVE a query I want it to remember what was the DB used for it.

Going further the table selector also does not make sense. I am quite likely to want to join multiple tables. I think we can really look at the Athena UI here to understand how it should work. I would implement it exactly like Athena UI does. DB is a dropdown and then just a component that lists all tables and lets you inspect them.

So TL;DR

1) Team and Environment should be part of worksheet creation and not modifiable
2) DB dropdown selected value should be remembered when saving queries
3) table selector should not be a dropdown and work like in Athena UI
4) add a download button to download result of data (could be a separate ticket idea + should be behind feature flag)
5) allow to have multiple queries in a single sheet like in Athena UI (separate ticket idea)
